packages/components/Menu/src/MenuList/useMenuList.ts

```diff
@@ -129,7 +170,27 @@ export const useMenuList = (_props: MenuListProps): MenuListState => {
     selectRadio,
     addRadioItem,
     removeRadioItem,
+    trackMenuItem,
+    untrackMenuItem,
+    onListKeyDown,
     hasMaxHeight: context.hasMaxHeight,
     hasMaxWidth: context.hasMaxWidth,
   };
 };
+
+export const useMenuItemTracking = (ref: React.RefObject<View>, disabled: boolean) => {
+  const { trackMenuItem, untrackMenuItem } = useMenuListContext();
+  const item = React.useMemo(
+    () => ({
+      ref,
+      disabled,
+    }),
+    [ref, disabled],
+  );
+
+  React.useEffect(() => {
+    trackMenuItem(item);
+    return () => untrackMenuItem(item);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
nit: I'd add a comment that this is intentional since you only want it to run once

Reply:
Thoughts on defining this as a separate hook `useMountEffect` so it's got a nice semantic name? 
https://stackoverflow.com/a/56767883

Reply:
sure, that seems fine to me.